src/js/stores/AppFormValidators.js

```diff
@@ -17,17 +17,18 @@ const AppFormValidators = {
       }
       return ValidConstraints.indexOf(p[1].toLowerCase()) !== -1;
     }),
-  env: (obj) =>
+  env: (obj) => obj.hasOwnProperty("key") && obj.hasOwnProperty("value") &&
```

Comment:
Yes, I also thought we said this once, but then I was confused to see it always with the append-way. Now I am more confused. :)
https://github.com/mesosphere/marathon-ui/pull/197/files#diff-546eb4eb9bf43cbe4b6adb02bdebe2b7R359